cmd/kubeadm/app/phases/bootstraptoken/clusterinfo/clusterinfo.go

```diff
@@ -40,16 +38,13 @@ const (
 )
 
 // CreateBootstrapConfigMapIfNotExists creates the kube-public ConfigMap if it doesn't exist already
-func CreateBootstrapConfigMapIfNotExists(client clientset.Interface, file string) error {
+func CreateBootstrapConfigMapIfNotExists(client clientset.Interface, kubeconfig *clientcmdapi.Config) error {
 
 	fmt.Printf("[bootstrap-token] Creating the %q ConfigMap in the %q namespace\n", bootstrapapi.ConfigMapClusterInfo, metav1.NamespacePublic)
 
-	klog.V(1).Infoln("[bootstrap-token] loading admin kubeconfig")
-	adminConfig, err := clientcmd.LoadFromFile(file)
-	if err != nil {
-		return errors.Wrap(err, "failed to load admin kubeconfig")
-	}
-	if err = clientcmdapi.FlattenConfig(adminConfig); err != nil {
+	// Clone the kubeconfig so that it's not mutated.
+	adminConfig := kubeconfig.DeepCopy()
+	if err := clientcmdapi.FlattenConfig(adminConfig); err != nil {
 		return err
 	}
 
```

Comment:
i think on this line to avoid a panic we should do a simple check and throw an error.

```
if adminConfig.Contexts[adminConfig.CurrentContext] == nil {
	return errors.New("invalid kubeconfig")
}
```

errors is `"github.com/pkg/errors"`